### PR TITLE
wcm 0.19.0.1 (new formula)

### PR DESCRIPTION
**Pull request by corporateshark**
Added Wal Commander file manager formula.

### CODE REVIEW

File: Library/Formula/wcm.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "formula"
```
Comment (DomT4) [inline Library/Formula/wcm.rb]: :scissors: this line.

File: Library/Formula/wcm.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class Wcm < Formula
+  homepage "http://wcm.linderdaum.com"
+  url "https://github.com/corporateshark/WalCommander/archive/release-0.19.0.1.zip"
+  sha1 "5a741835233d832923578d1633f4fda6fee4868d"
+
+  depends_on 'cmake' => :build
```
Comment (DomT4) [inline Library/Formula/wcm.rb]: Double quotes everywhere.

File: Library/Formula/wcm.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  depends_on "freetype"
+  depends_on "libssh2"
+  depends_on :x11
+
+  def install
+    system "cmake -G \"Unix Makefiles\""
+    system "make all -j16 -B"
```
Comment (DomT4) [inline Library/Formula/wcm.rb]: ` system "make", "all", "-j16", "-B" ` ideally.

File: Library/Formula/wcm.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    system "make install"
```
Comment (DomT4) [inline Library/Formula/wcm.rb]: ` system "make", "install" `.

**Comment by DomT4** [comment]
The hard X11 dep for GUI purposes may make this a better fit for [Homebrew/homebrew-x11](https://github.com/Homebrew/homebrew-x11).

File: Library/Formula/wcm.rb (hunk 3 above)
Comment (xu-cheng) [inline Library/Formula/wcm.rb]: I think it should be `system "make", "all", "-j#{ENV.make_jobs}", "-B"`

Comment (DomT4) [inline Library/Formula/wcm.rb]: Yeah, presuming the ` -j ` pertains to make jobs here, it should be.

File: Library/Formula/wcm.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  url "https://github.com/corporateshark/WalCommander/archive/release-0.19.0.1.zip"
+  sha1 "5a741835233d832923578d1633f4fda6fee4868d"
+
+  depends_on 'cmake' => :build
+  depends_on "freetype"
+  depends_on "libssh2"
+  depends_on :x11
+
+  def install
+    system "cmake -G \"Unix Makefiles\""
```
Comment (xu-cheng) [inline Library/Formula/wcm.rb]: ` system "cmake", ".", *std_cmake_args`

**Comment by corporateshark** [comment]
Is there a way to disable ``mountain_lion`` builds? We don't support it anyway.

**Comment by DomT4** [comment]
` depends_on :macos => :mountain_lion ` would do the trick.

File: Library/Formula/wcm.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "freetype"
+  depends_on "libssh2"
+  depends_on :x11
+  depends_on :macos => :mavericks
+
+  needs :cxx11
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "all", "-j#{ENV.make_jobs}", "-B"
```
Comment (mikemcquaid) [inline Library/Formula/wcm.rb]: Pretty sure the `-j` will be done without you specifying it.

File: Library/Formula/wcm.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/wcm", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/wcm.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/wcm.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+class Wcm < Formula
+  homepage "http://wcm.linderdaum.com"
+  url "https://github.com/corporateshark/WalCommander/archive/release-0.19.0.1.tar.gz"
+  sha256 "af2b0779e7ffa51fad0c3901a8bd256546bc1bb7f4f156087f9706b066f8a871"
+
+  depends_on "cmake" => :build
+  depends_on "freetype"
+  depends_on "libssh2"
+  depends_on :x11
```
Comment (mikemcquaid) [inline Library/Formula/wcm.rb]: This should be submitted to `homebrew-x11` or this dependency made optional (if possible).

Comment (DomT4) [inline Library/Formula/wcm.rb]: https://github.com/Homebrew/homebrew-x11/pull/38

File: Library/Formula/wcm.rb (hunk 7 above)
Comment (corporateshark) [inline Library/Formula/wcm.rb]: Probably not. This is a GUI application.

File: Library/Formula/wcm.rb (hunk 8 above)
Comment (corporateshark) [inline Library/Formula/wcm.rb]: The x11 dependency is hardwired.

**Comment by tdsmith** [comment]
Closing in favor of x11 PR.

**Comment by corporateshark** [comment]
@IlyaSemenov Fixed SHA256 in https://github.com/Homebrew/homebrew-x11/pull/38

**Pull request closed without merging**